sockeye/inference.py

```diff
@@ -1778,6 +1877,12 @@ def _beam_search(self,
         # (0) encode source sentence, returns a list
         model_states = self._encode(source, source_length)
 
+        # extract predicted length ratio from model states (if present), average them over models 
+        # and return the predicted reference length
+        model_states, estimated_reference_lengths = self._extract_length_ratios(model_states,
```

Comment:
I'd prefer if the 'extraction' would be done directly in the code that runs the module and gets the outputs. Basically hide these internals from the beam search code here a little bit.